src/comparator.rs

```diff
@@ -0,0 +1,826 @@
+                comp: $COMP,
+                positive_input: P,
+                negative_input: N,
+                config: Config,
+                clocks: &Clocks,
+            ) -> Self {
+                comp.comparator(positive_input, negative_input, config, clocks)
+            }
+
+            /// Enables the comparator
+            pub fn enable(self) -> Comparator<$COMP, Enabled> {
+                self.regs.csr().modify(|_, w| w.en().set_bit());
+                Comparator {
+                    regs: self.regs,
+                    _enabled: PhantomData,
+                }
+            }
+
+            /// Enables raising the `ADC_COMP` interrupt at the specified output signal edge
+            pub fn listen(&self, edge: SignalEdge, exti: &EXTI) {
+                exti.listen($Event, edge);
+            }
+        }
+
+        impl<ED: EnabledState> Comparator<$COMP, ED> {
+            /// Returns the value of the output of the comparator
+            pub fn output(&self) -> bool {
+                self.regs.csr().read().value().bit_is_set()
+            }
+        }
+
+        impl Comparator<$COMP, Enabled> {
+            pub fn lock(self) -> Comparator<$COMP, Locked> {
+                // Setting this bit turns all other bits into read only until restart
+                self.regs.csr().modify(|_, w| w.lock().set_bit());
+                Comparator {
+                    regs: self.regs,
+                    _enabled: PhantomData,
+                }
+            }
+
+            /// Disables the comparator
+            pub fn disable(self) -> Comparator<$COMP, Disabled> {
+                self.regs.csr().modify(|_, w| w.en().clear_bit());
+                Comparator {
+                    regs: self.regs,
+                    _enabled: PhantomData,
+                }
+            }
+        }
+
+        impl<ED> Comparator<$COMP, ED> {
+            /// Disables raising interrupts for the output signal
+            pub fn unlisten(&self, exti: &EXTI) {
+                exti.unlisten($Event);
+            }
+
+            /// Returns `true` if the output signal interrupt is pending for the `edge`
+            pub fn is_pending(&self, exti: &EXTI) -> bool {
+                exti.is_pending($Event)
+            }
+
+            /// Unpends the output signal interrupt
+            pub fn unpend(&self, exti: &EXTI) {
+                exti.unpend($Event);
+            }
+
+            /// Configures a GPIO pin to output the signal of the comparator
+            ///
+            /// Multiple GPIO pins may be configured as the output simultaneously.
+            pub fn output_pin<P: OutputPin<$COMP>>(&self, pin: P) {
+                pin.setup();
+            }
+        }
+    };
+}
+
+impl_comparator!(COMP1, comp1, ExtiEvent::COMP1);
+impl_comparator!(COMP2, comp2, ExtiEvent::COMP2);
+
+/*
+/// Uses two comparators to implement a window comparator.
+/// See Figure 69 in RM0444 Rev 5.
+pub struct WindowComparator<U, L, ED> {
```

Comment:
Same thing with window mode, not supported

Reply:
Also remove every inapplicable code.